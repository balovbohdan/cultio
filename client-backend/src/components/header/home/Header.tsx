import * as React from 'react';
import * as classnames from 'classnames';
import {Grid, Paper, Button} from '@material-ui/core';

import * as logo from '@img/logo.svg';

import {useCss} from './style';

export const Header = () => {
    const css = useCss();

    return (
        <div className={css.root}>
            <Grid className={css.container} container spacing={1}>
                <Logo css={css}/>
                <Actions css={css}/>
            </Grid>
        </div>
    );
};

const Logo = ({css}) =>
    <Grid className={css.item} item xs={1}>
        <Paper className={css.paper}>
            <img
                alt='Logo'
                src={logo}
                className={css.logoImg}/>
        </Paper>
    </Grid>;

const Actions = ({css}) => {
    const classNamePaper = classnames(css.paper, css.paperActions);

    return (
        <Grid className={css.item} item xs={8}>
            <Paper className={classNamePaper}>
                <Button className={css.btn}>About</Button>
                <Button className={css.btn}>Support</Button>
                <RegisterBtn css={css}/>
                <SignInBtn css={css}/>
            </Paper>
        </Grid>
    );
};

const SignInBtn = ({css}) =>
    <Button variant='outlined' color='primary' className={css.btn}>
        Sign In
    </Button>;

const RegisterBtn = ({css}) =>
    <Button variant='contained' color='secondary' className={css.btn}>
        Register
    </Button>;