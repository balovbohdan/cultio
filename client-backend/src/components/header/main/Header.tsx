import * as React from 'react';

import {Menu} from '@material-ui/icons';
import {makeStyles} from '@material-ui/core/styles';

import {
    AppBar,
    IconButton,
    Button,
    Typography,
    Toolbar
} from '@material-ui/core';

const useStyles = makeStyles(theme => ({
    root: {
        flexGrow: 1
    },
    menuButton: {
        marginRight: theme.spacing(2)
    },
    title: {
        flexGrow: 1
    }
}));

export const Header = () => {
    const styles = useStyles();

    return (
        <>
            <div className={styles.root}>
                <AppBar position='static'>
                    <Toolbar>
                        <IconButton
                            edge='start'
                            className={styles.menuButton}
                            color='inherit'
                            aria-label='menu'>
                            <Menu/>
                        </IconButton>
                        <Typography variant='h6' className={styles.title}>
                            News
                        </Typography>
                        <Button color='inherit'>Login</Button>
                    </Toolbar>
                </AppBar>
            </div>
        </>
    );
};