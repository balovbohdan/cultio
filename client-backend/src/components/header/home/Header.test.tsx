import * as React from 'react';
import {shallow} from 'enzyme';

import {Header} from './Header';

describe('header/home/Header', () => {
    it('Should render correctly', () => {
        const header = shallow(<Header/>);

        expect(header).toMatchSnapshot();
    });
});