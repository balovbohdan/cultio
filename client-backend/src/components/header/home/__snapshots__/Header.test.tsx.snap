// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header/home/Header Should render correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-container-2"
    container={true}
    spacing={1}
  >
    <Logo
      css={
        Object {
          "btn": "makeStyles-btn-7",
          "container": "makeStyles-container-2",
          "item": "makeStyles-item-3",
          "logoImg": "makeStyles-logoImg-4",
          "paper": "makeStyles-paper-5",
          "paperActions": "makeStyles-paperActions-6",
          "root": "makeStyles-root-1",
        }
      }
    />
    <Actions
      css={
        Object {
          "btn": "makeStyles-btn-7",
          "container": "makeStyles-container-2",
          "item": "makeStyles-item-3",
          "logoImg": "makeStyles-logoImg-4",
          "paper": "makeStyles-paper-5",
          "paperActions": "makeStyles-paperActions-6",
          "root": "makeStyles-root-1",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
</div>
`;
